### DISCUSSION TITLE
[Guide] When to use Discussions, Issues, or Slack

### DISCUSSION THREAD

**Discussion by hf200012** [apache/doris#14161] [RFC]
Welcome on board ! We are using Disuccsions as a platform to fostering connectos within the Doris community, specially for addressing technical inquiries and providing support. 

For more extensive questions, reports, or development-related topics, please use 🔗[Issues](https://github.com/apache/doris/issues) 
For real-time communications and online meetings, we invite you to join 🔗[Slack community](https://join.slack.com/t/apachedoriscommunity/shared_invite/zt-1x7x8fger-F7NoshFQn~djlvGdnEtxUQ).

We encourage you to use discussions for the following purpose:
- Talking about user issues, questions you come across
- Sharing your great ideas
- Requesting or presenting proposals for innovative features

To best assit in resolving and discussing peobems, here are some templates for posting questions when or reporting bugs: 

## Error Description

1. Describe what operation has caused the error. For example:

```
Execute SQL: 
select * from tbl limtit 20 

The following error is returned:
ERROR 1105 (HY000): RpcException, msg: io.grpc.StatusRuntimeException: UNAVAILABLE: Network closed for unknown reason
```

2. Provide details about the version of Doris that you're currently running

```
Execute "show backends; show frontends" to check BE and FE versions

FE ：doris-1.2.0-rc01-636faec
BE ：doris-1.2.0-rc01-636faec

You can also check backend by executing "./lib/doris_be --version"

Example:
./doris_be --version
doris-1.2.0-unstable-rc01 RELEASE (build git://d3f88ac40e6b@3fff8f399baad5f060a1992447a854c165f203f2)
Built on Thu, 10 Nov 2022 06:28:56 UTC by d3f88ac40e6b

This is to confirm that you are not using different versions of backend and frontend.
```

3. Share information about your cluster

4. It would be helpful to provide details about the FE nodes and their configurations. Additionally, if you can provide the fe.conf file, it will assist in troubleshooting the issue.

5. Please provide details about the FE nodes, including memory and CPU specifications. Additionally, providing the be.conf file for further analysis would be beneficial.

6. If you have created a GitHub issue related to this error, please consider sharing the link for reference.

## FE Troubleshooting

Common problems:

- Query deadlock
- High CPU usage
- High memory usage
- Abnormal FE termination

To solve these problems, we need the following information for troubleshooting:

1. For abnormal FE termination, we will first check the fe.out and then the fe.log, so please package these files and provide them.
2. Check FE stack via a jstack command.

```
jstack <fe pid> > fe.jstack
```

3. In the case of high memory usage, please use the jmap command to print relevant information.

```
jmap -histo:live <fe pid>  > jmap.text
```

4. Please provide full fe.log of the problematic FE

## BE Troubleshooting

To solve BE outage, follow these steps and provide the relevant log information:

1. Execute dmesg -T to see if there is an OOM error.

```
[Thu Nov 10 11:57:36 2022] oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/user.slice/user-1000.slice/session-2065.scope,task=doris_be,pid=245473,uid=1000
[Thu Nov 10 11:57:36 2022] Out of memory: Killed process 245473 (doris_be) total-vm:86515296kB, anon-rss:62635568kB, file-rss:0kB, shmem-rss:0kB, UID:1000 pgtables:144056kB oom_score_adj:0
[Thu Nov 10 11:57:38 2022] oom_reaper: reaped process 245473 (doris_be), now anon-rss:0kB, file-rss:0kB, shmem-rss:0kB
```

If there is, check the timestamp to see if it is when you encountered the error. If it does, that means OOM is the root cause. You can fix it by restarting the BE.

2. If there is no OOM error, check the be.out file and see if there exists stack information.

- If you have used Supervisor to auto-restart processes, the stack information in be.out will be overwritten by the Supervisor log. In this case, you need to find the relevant stack information in the logs in Supervisor.

- If you encounter this problem after executing a certain SQL, please provide the following information:

  i. The SQL statement

  ii. The corresponding table creation statement

- If it were not BE Core issue caused by the SQL that is running, follow these steps:

  i. [Open the CoreDump file and locate the problematic SQL](https://zhuanlan.zhihu.com/p/555309629)

  ii. Provide the SQL and the relevant table creation statement

## Query Error or Other Problems

If it is not a FE or BE issue but a problem that occurs during your SQL operations (table creation, query execution) or data compaction, please provide the following information:

1. Error message that is returned
2. fe.log
3. be.INFO
4. A screenshot of Grafana dashboard for the past hour (if you have installed Grafana)

——————————————————

## 问题描述
1. 描述这个问题是怎么出现的，做了什么操作
```
比如：执行 SQL 
select * from tbl limtit 20 
查询返回下面的错误:
ERROR 1105 (HY000): RpcException, msg: io.grpc.StatusRuntimeException: UNAVAILABLE: Network closed for unknown reason
```
2. 使用的 Doris 版本 ,这里要详细的信息,最好是到commit
```
用户可以通过show backends; show frontends 拿到 fe 和 be 的版本信息
FE ：doris-1.2.0-rc01-636faec
BE ：doris-1.2.0-rc01-636faec
BE 可以通过执行./lib/doris_be --version 查看
例如：
./doris_be --version
doris-1.2.0-unstable-rc01 RELEASE (build git://d3f88ac40e6b@3fff8f399baad5f060a1992447a854c165f203f2)
Built on Thu, 10 Nov 2022 06:28:56 UTC by d3f88ac40e6b

这里这样做的目的是为了，排除 fe 和 be 使用不同的版本
```
3. 集群基本信息
1. FE 的节点数及配置（最好将fe.conf配置文件一块提供）
2. BE 的节点数及配置（包括内存、CPU、be.conf 配置文件）
3. 如果用户有记录github issues，将github issues 一块提供

## FE 问题定位
FE 常见问题：

- 查询卡死
- CPU 占用率过高
- 内存占用过高
- FE 异常退出

针对 FE 上面问题，出现上面这些情况我们需要用户提供一下信息
1. fe异常退出，也是先看 fe.out，再看 fe.log，将 fe.out 和 fe.log  日志打包发给我们
2. 通过 jstack 命令拿到 fe的堆栈信息
```
jstack <fe pid> > fe.jstack
```
3. 内存占用过高问题，使用 jmap 打印相关信息
```
jmap -histo:live <fe pid>  > jmap.text
```
5. 日志（ fe.log ）
提供出现问题点的fe.log 完整日志文件打包，一块提供
## BE 问题
一般是 BE 挂掉，需要用户按照如下排查并提供相应日志信息
1. 首先需要用户执行 dmesg -T  看看有没有OOM信息
```
[Thu Nov 10 11:57:36 2022] oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/user.slice/user-1000.slice/session-2065.scope,task=doris_be,pid=245473,uid=1000
[Thu Nov 10 11:57:36 2022] Out of memory: Killed process 245473 (doris_be) total-vm:86515296kB, anon-rss:62635568kB, file-rss:0kB, shmem-rss:0kB, UID:1000 pgtables:144056kB oom_score_adj:0
[Thu Nov 10 11:57:38 2022] oom_reaper: reaped process 245473 (doris_be), now anon-rss:0kB, file-rss:0kB, shmem-rss:0kB
```
如果出现下面的这些信息，核对时间点是否符合，如果符合说明是内存不足，直接重启 be
2. 如果不是OOM，让用户查看 be.out 文件，是否有堆栈信息

- 如果用户使用了 supervisor 实现进程自动拉起的，be.out 里的堆栈信息会被supervisor的日志吞吃掉，需要查看supervisor里的日志，找到对应的堆栈，并提供出来
- 如果是用户手动在执行某个 SQL 出现这个问题，让用户提供一下信息
    1. 查询 SQL 语句
    2. 这个 SQL 语句对应的建表语句

- 如果不是手动正在执行的SQL语句引发的 BE Core 问题，让用户按照下面的方法操作

    1. [如何打开CoreDump文件及快速定位SQL](https://zhuanlan.zhihu.com/p/555309629) 
    2. 然后拿到出问题的SQL及这个SQL 对应的建表语句
## 查询错误或者其他的问题
如果不是上面这些问题，而是用户在执行SQL操作（建表、查询等操作），Compactoion问题等
用户需要提供以下信息：
1. 正在执行操作返回的异常信息
3. FE的日志（fe.log）
4. 对应 BE 节点的日志（be.INFO）
5. 如果安装了 Grafana 监控，同时提供至少 1 个小时的监控截图信息

**Comment by LoganCjn** [comment]
创建Hive外表时报错：
faild to check if table is exists int XXX.db,org.apache.thrift. TApplicasionException: Invalid method name:  ‘get_table_reg‘’

**Reply by LemonLiTree** [reply]
specify the version： 'hive.version' = '1.1.0' in properties

**Comment by MrRightLi** [comment]
1. 我就想问为什么按照官方文档，无法构建docker镜像，下载的是 tar.xz,为什么文档说的tar.gz.
2. 还有dockerfile里面没提解压tar文件呀？
3. 从入门到放弃？

**Reply by FreeOnePlus** [reply]
1. 在最开始社区提供的是 tar.gz 格式的压缩包，后来由于基金会要求，拆分为三个包，且压缩格式更改为 tar.xz，如果要进行 DockerImage 构建，需要手动整理一下官方的包，重新 tar -zcvf *.tar.gz，这一块后续我会更新到官网
2. 建议谷歌一下关于 Dockerfile 里添加压缩包的两种命令，具体使用含义：ADD & COPY
3. 如果想直接 Docker 启动，可以在 Docker 部署章节直接使用官方二进制包

**Comment by gagan211** [comment]
i am having a CSV in which the  column named ["log_time"] has the data in the form of 2024-01-01 091500.000000.
the DORIS database table schema also has the column named "log_time" and the Datatype is of DATETIME 
but still when i load data from the csv to the DB.Table 
it will only load the date and not the time so it will be like "2024-01-01 00:00:00" can anyone help me over here please ?
